runtime/src/main/java/org/corfudb/runtime/LogReplicationServiceClient.java

```diff
@@ -0,0 +1,339 @@
+     */
+    public LogReplicationServiceClient(CorfuRuntime runtime, String clientName) throws InvocationTargetException, NoSuchMethodException, IllegalAccessException {
+        this.corfuStore = new CorfuStore(runtime);
+        this.clientName = clientName;
+
+        this.modelEnum = "LOGICAL_GROUPS";
+
+        this.clientKey = ClientRegistrationId.newBuilder()
+                .setClientName(clientName)
+                .build();
+        this.clientInfo = ClientRegistrationInfo.newBuilder()
+                .setClientName(clientName)
+                .setModel(LogReplication.ReplicationModel.valueOf(modelEnum))
+                .setRegistrationTime(Timestamp.newBuilder().setSeconds(new Date().getTime() / 1000).build())
+                .build();
+
+        this.replicationRegistrationTable = corfuStore.openTable(CORFU_SYSTEM_NAMESPACE, LR_REGISTRY_TABLE_NAME, ClientRegistrationId.class,
+                ClientRegistrationInfo.class, null, TableOptions.builder().build());
+        this.sourceMetadataTable = corfuStore.openTable(CORFU_SYSTEM_NAMESPACE, LR_SOURCE_METADATA_TABLE_NAME, ClientDestinationInfoKey.class,
+                DestinationInfoVal.class, null, TableOptions.builder().build());
+
+        register();
+    }
+
+    /**
+     * Adds client and model to LogReplicationRegistrationTable
+     */
+    private void register() {
+        try {
+            IRetry.build(ExponentialBackoffRetry.class, () -> {
+                try (TxnContext txn = corfuStore.txn(CORFU_SYSTEM_NAMESPACE)) {
+                    if (txn.isExists(LR_REGISTRY_TABLE_NAME, clientKey)) {
+                        log.warn("Client already registered \nclientKey: {} \nclientInfo: {}",
+                                clientKey, txn.getRecord(replicationRegistrationTable, clientKey).getPayload().toString());
+                    }
+                    else {
+                        txn.putRecord(replicationRegistrationTable, clientKey, clientInfo, null);
+                        txn.commit();
+                    }
+                }
+                catch (Exception e) {
+                    log.error("Unable to register client {}, Error: {}", clientName, e);
+                }
+                return null;
+            }).setOptions(x -> x.setMaxRetryThreshold(Duration.ofSeconds(SYNC_THRESHOLD))).run();
+        } catch (InterruptedException e) {
+            throw new UnrecoverableCorfuInterruptedError(e);
+        }
+    }
+
+    /**
+     * Add destination to LogReplicationSourceMetadataTable for monitoring
+     *
+     * @param logicalGroup String representative of the group of tables to be replicated
+     * @param destination Cluster which the logicalGroup should be replicated to
+     * @return true if destination was newly registered
+     */
+    public boolean addDestination(String logicalGroup, String destination) {
+        if (Strings.isNullOrEmpty(logicalGroup) || Strings.isNullOrEmpty(destination)) {
```

Comment:
Consider using Preconditions.checkArgument() as it will throw an exception in such a case.

Example: https://github.com/CorfuDB/CorfuDB/blob/master/corfudb-tools/src/main/java/org/corfudb/browser/CorfuStoreBrowserEditorMain.java#L301